src/commands/UserUpsertCommand.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  MultisigUserUpdateMessage,
+  USER_UPDATE_MESSAGE_SCHEMA,
+} from "../lib/users/schemas.js";
+import { isTypedMessage } from "../utils/signatures.js";
+import UserUpsertSignatureVerifier from "../lib/safe-signature-verification/UserUpsertSignatureVerifier.js";
+import { Database } from "../types/supabaseData.js";
+
+import { SafeApiCommand } from "./SafeApiCommand.js";
+
+type SignatureRequest =
+  Database["public"]["Tables"]["signature_requests"]["Row"];
+
+export class UserUpsertCommand extends SafeApiCommand {
+  async execute(): Promise<void> {
+    const signatureRequest = await this.dataService.getSignatureRequest(
+      this.safeAddress,
+      this.messageHash,
+    );
+
+    if (!signatureRequest || signatureRequest.status !== "pending") {
+      return;
+    }
+
+    const safeMessage = await this.safeApiKit.getMessage(this.messageHash);
+
+    if (!isTypedMessage(safeMessage.message)) {
+      throw new Error("Unexpected message type: not EIP712TypedData");
```

Comment:
there's no log here, but there is one a few lines lower. How will the error be caught? Will it be logged there?

Reply:
It would be caught here: https://github.com/pheuberger/hypercerts-api/blob/d4eb5845991ed5c09e12819660d9463e39fbf9a1/src/services/SafeApiQueue.ts#L87-L92

Since the error here doesn't provide any additional information, it's fine to delegate it to the surrounding try catch to log. The case a few lines down has extra information, so I also logged it here. I could've also created a custom error that wraps this parse error and unwrap that inside of the queue, but felt that it wasn't much better either.